host/contracts/manager.go

```diff
@@ -309,18 +313,21 @@ func (cm *Manager) RenewV2Contract(renewal rhp4.TransactionSet, usage proto4.Usa
 	fc := resolution.NewContract
 
 	// sanity checks
+	refresh := fc.ExpirationHeight == existing.ExpirationHeight
```

Comment:
I'd get rid of all of these and handle it in coreutils. The rhp4 server should be handling these kinds of validations.